ui/imports/shared/views/chat/MessageView.qml

```diff
@@ -1170,16 +1172,51 @@ Loader {
 
     Component {
         id: profileContextMenuComponent
-
         ProfileContextMenu {
             store: root.rootStore
+
             onOpenProfileClicked: (publicKey) => {
+                console.log("MessageView.qml:1: open profile clicked", publicKey)
                 Global.openProfilePopup(publicKey, null)
             }
             onCreateOneToOneChat: (communityId, chatId, ensName) => {
                 Global.changeAppSectionBySectionType(Constants.appSection.chat)
                 root.rootStore.chatCommunitySectionModule.createOneToOneChat("", chatId, ensName)
             }
+            onReviewContactRequest: (publicKey) => {
+                const contactDetails = publicKey === "" ? {} : Utils.getContactDetailsAsJson(publicKey, true, true)
```

Comment:
WDYT of using the `ContactDetails` component as a helper for the `ProfileContextMenu` instead of `Utils.getContactDetailsAsJson` usages. I think it would simplify our life quite a bit.
We'd end-up with something like this:

```
Global.openMenu(profileContextMenuComponent, sender, { selectedUserPublicKey: publicKey })
...

ProfileContextMenu {
    ContactDetails {
                id: contactDetails
                publicKey: selectedUserPublicKey
                contactsStore: root.contactsStore
                profileStore: root.profileStore
         }
         
         selectedUserDisplayName: contactDetails.displayName
         selectedUserIcon: contactDetails.icon
          profileType: Utils.getProfileType(publicKey, myPublicKey, isBridgedAccount, isBlocked);
          contactType: Utils.getContactType(contactDetails.contactRequestState, contactDetails.isContact);
          trustStatus: contactDetails.trustStatus
          ensVerified: contactDetails.ensVerified
          onlineStatus: contactDetails.onlineStatus
          hasLocalNickname: !!contactDetails.localNickname
    }
```

Would be a step in the right direction in the sense that we won't need to revisit this code when removing usages of `Utils.getContactDetailsAsJson`. Another big advantage IMO is that it completely removes the need of adding more dependencies to the nim layer in the Utils singleton (Utils.getProfileContext).

Reply:
not sure I understood what you want here, may I suggest in a different PR though?

Reply:
sounds good to me! The idea here is to use `ContactDetails` instead of `getContactDetailsAsJson` where possible.